frontend/src/Components/UserProfileSettings.tsx

```diff
@@ -27,12 +29,41 @@ export default function UserProfileSettings(props: UserProfileSettingsProps) {
 
   let gender = props.gender;
 
-  const handleLogout = (e: React.MouseEvent) => {
-    e.preventDefault();
-    api.auth.signOut().then(() => {
-      navigate(location.pathname);
-    });
-  };
+    const confirmWrapper = (message: string, callback: () => void) => (e: React.MouseEvent) => {
+        e.preventDefault();
+        confirmAlert({
+            title: 'Астанавитесь! Подумайте!',
+            message,
+            buttons: [
+                {
+                    label: 'Да!',
+                    onClick: callback
+                },
+                {
+                    label: 'Отмена',
+                    className: 'cancel'
+                }
+            ],
+            overlayClassName: 'orbitar-confirm-overlay'
+        });
+    };
+
+  const handleLogout = confirmWrapper(
+      'Вы действительно хотите выйти? Текущая сессия будет сброшена. Вы будете вынуждены войти в аккаунт заново.', () => {
```

Comment:
nit. Too specific imho.
```suggestion
      'Вы действительно хотите выйти? Вы будете вынуждены войти в аккаунт заново.', () => {
```